includes/Classifai/Admin/BulkActions.php

```diff
@@ -76,29 +110,84 @@ public function bulk_action_handler( $redirect_to, $doaction, $post_ids ) {
 		foreach ( $post_ids as $post_id ) {
 			$this->save_post_handler->classify( $post_id );
 		}
+		$redirect_to = remove_query_arg( [ 'bulk_classified', 'bulk_scanned', 'bulk_cropped' ], $redirect_to );
 		$redirect_to = add_query_arg( 'bulk_classified', count( $post_ids ), $redirect_to );
 		return $redirect_to;
 	}
 
 	/**
-	 * Display an admin notice after classifying posts in bulk.
+	 * Handle media bulk actions.
+	 *
+	 * @param string $redirect_to       Redirect URL after bulk actions.
+	 * @param string $doaction          Action ID.
+	 * @param array  $attachment_ids    Attachment ids to apply bulk actions to.
+	 *
+	 * @return string
+	 */
+	public function media_bulk_action_handler( $redirect_to, $doaction, $attachment_ids ) {
+		if (
+			empty( $attachment_ids ) ||
+			! in_array( $doaction, [ 'scan_image', 'smart_crop' ], true )
+		) {
+			return $redirect_to;
+		}
+
+		foreach ( $attachment_ids as $attachment_id ) {
+			$current_meta = wp_get_attachment_metadata( $attachment_id );
+
+			if ( 'smart_crop' === $doaction ) {
+				$this->computer_vision->smart_crop_image( $current_meta, $attachment_id );
+			} else {
+				$this->computer_vision->generate_image_alt_tags( $current_meta, $attachment_id );
+			}
+		}
+
+		$action = 'scan_image' === $doaction ? 'scanned' : 'cropped';
+
+		$redirect_to = remove_query_arg( [ 'bulk_classified', 'bulk_scanned', 'bulk_cropped' ], $redirect_to );
+		$redirect_to = add_query_arg( "bulk_{$action}", count( $attachment_ids ), $redirect_to );
+		return $redirect_to;
+	}
+
+	/**
+	 * Display an admin notice after bulk updates.
 	 */
 	public function bulk_action_admin_notice() {
-		if ( empty( $_REQUEST['bulk_classified'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+
+		$classified = filter_input( INPUT_GET, 'bulk_classified', FILTER_SANITIZE_NUMBER_INT );
+		$scanned    = filter_input( INPUT_GET, 'bulk_scanned', FILTER_SANITIZE_NUMBER_INT );
+		$cropped    = filter_input( INPUT_GET, 'bulk_cropped', FILTER_SANITIZE_NUMBER_INT );
```

Comment:
Is it possible to use the WP sanitization functions here? 

`$thing = ! empty( $_GET['thing'] ) ? (int) wp_unslash( $_GET['thing'] ) : 0;`